api-report/tree2.api.md

```diff
@@ -1377,10 +1377,11 @@ type MapFieldSchema = FieldSchema<typeof FieldKinds.optional | typeof FieldKinds
 // @alpha
 export interface MapNode<TSchema extends MapSchema> extends TreeNode {
     // (undocumented)
-    [Symbol.iterator](): Iterator<TypedField<TSchema["mapFields"]>>;
+    [Symbol.iterator](): IterableIterator<UnboxField<TSchema["mapFields"], "notEmpty">>;
     readonly asObject: {
         readonly [P in FieldKey]?: UnboxField<TSchema["mapFields"]>;
     };
+    boxedIterator(): IterableIterator<TypedField<TSchema["mapFields"]>>;
```

Comment:
@noencke - unboxed iteration is good, but similar to the previous PR, our goal is to avoid exposing boxing unless we find ourselves wedged with no other options.  I'd suggest hiding this iterator behind a symbol for now, and maybe marking it as deprecated out of paranoia.

Reply:
Okay, I've created a "boxedIterator" symbol.